cmd/krel/cmd/automate.go

```diff
@@ -0,0 +1,223 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/util/homedir"
+
+	"k8s.io/release/pkg/automate"
+)
+
+const (
+	k8sOrg string = "kubernetes"
+	websiteRepo string = "website"
+)
+
+type automateOptions struct {
+	githubToken string
+	credFile string
+	sheetId string
+	sheetName string
+	rangeCol string
+	colID []string
+	release string
+}
+
+type PrData struct {
+	KubernetesPR int
+	KEP int
+	WebsitePR int
+	Assignee string
+	Sig string
+}
+
+var automateOpts = automateOptions{
+	credFile:  getCredentialFile(),
+	sheetName: "Docs",
+	rangeCol:  "A1:K",
+	colID: []string{"A", "G", "I", "J", "K"},
+}
+
+// announceCmd represents the subcommand for `krel announce`
+var automateCmd = &cobra.Command{
+	Use:           "automate",
+	Short:         "Automate some work for release docs",
+	SilenceUsage:  true,
+	SilenceErrors: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return cmd.Help()
+	},
+}
+
+func init() {
+	automateCmd.PersistentFlags().StringVarP(
+		&automateOpts.githubToken,
+		"github-token",
+		"",
+		"",
+		"github token for authentication",
+	)
+
+	automateCmd.PersistentFlags().StringVarP(
+		&automateOpts.credFile,
+		"cred-file",
+		"",
+		automateOpts.credFile,
+		"credentials for the Google Sheets API")
+
+	automateCmd.PersistentFlags().StringVarP(
+		&automateOpts.sheetId,
+		"sheet-id",
+		"",
+		"",
+		"Sheet ID of the spreadsheet, normally found in the url of the spreadsheet")
+
+	automateCmd.PersistentFlags().StringVarP(
+		&automateOpts.sheetName,
+		"name",
+		"",
+		automateOpts.sheetName,
+		"The name of the particular sheet that we want to read from")
+
+	automateCmd.PersistentFlags().StringVarP(
+		&automateOpts.rangeCol,
+		"range",
+		"",
+		automateOpts.rangeCol,
+		"The range of columns from which we want to read from e.g A1:B")
+
+	automateCmd.PersistentFlags().StringArrayVar(
+		&automateOpts.colID,
+		"col-ids",
+		automateOpts.colID,
+		// TODO(SomtochiAma): Better explanation
+		"the letters of the columns that contain the kep, k/w, sigs, assignee and k/k pr information")
+
+	automateCmd.PersistentFlags().StringVarP(
+		&automateOpts.release,
+		"release",
+		"",
+		"",
+		"The kubernetes release e.g 1.20")
+
+	rootCmd.AddCommand(automateCmd)
+}
+
+func getPrStruct(sheetID string, readRange string, headers []string) ([]*PrData, error) {
+	if len(headers) == 4 {
+		return nil, fmt.Errorf("column headers should be 5")
```

Comment:
```suggestion
		return nil, errors.New("column headers should be 5")
```
(same for other occurrences)